pages/marketplace/usage-analytics/responses.vue

```diff
@@ -0,0 +1,347 @@
+
+                    if (label) {
+                        label += ': ';
+                    }
+                    label += context.formattedValue + ' responses';
+                    return label;
+                },
+            },
+        },
+    },
+};
+
+const lineChartRefs = ref<any[]>([]);
+
+const setLineChartRef = (el: any, index: number) => {
+    if (el) {
+        lineChartRefs.value[index] = el;
+    }
+};
+
+const zoomInLineChart = (index: number) => {
+    // Access the specific chart instance from the array of refs
+    const chartInstance = lineChartRefs.value[index]?.chart;
+    if (chartInstance) {
+        chartInstance.zoom(1.1); // Zoom in by 10%
+    }
+};
+
+const zoomOutLineChart = (index: number) => {
+    const chartInstance = lineChartRefs.value[index]?.chart;
+    if (chartInstance) {
+        chartInstance.zoom(0.9); // Zoom out by 10%
+    }
+};
+
+const resetLineChartZoom = (index: number) => {
+    const chartInstance = lineChartRefs.value[index]?.chart;
+    if (chartInstance) {
+        chartInstance.resetZoom();
+    }
+};
+
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    BarElement,
+    Title,
+    Tooltip,
+    Legend,
+    PointElement,
+    LineElement,
+    ArcElement,
+    zoomPlugin,
+);
+
+const {
+    data: questionnaire,
+    status: questionnaireStatus,
+    error: questionnaireError,
+} = await useAsyncData<Questionnaire>(`fetchQuestionnaire-asset-${assetId.value}`, () =>
+    $fetch(`/api/usage-analytics/${assetId.value}`),
+);
+
+const { data, status, error } = useFetch<QuestionResponse[]>(`/api/usage-analytics/get-answers`, {
+    query: {
+        assetId: assetId.value,
+        forVerifiedBuyers: forVerifiedBuyers.value,
+    },
+});
+
+const filteredData = computed(() =>
+    data.value ? data.value?.filter((response: QuestionResponse) => response.questionType !== 'Text') : [],
+);
+
+const computedChartData = computed(() =>
+    filteredData.value
+        ? filteredData.value.map((response: QuestionResponse) => {
+              const labels = response.options;
+              const label = response.questionTitle;
+              const allTimeData: number[] = [];
+              const chartType = response.questionType === 'Checkbox' ? 'bar' : 'pie';
+
+              //All time data
+              labels?.forEach((label: string) => {
+                  allTimeData.push(
+                      response.responses.reduce((acc: number, curr: { response: string; date: string }) => {
+                          if (curr.response === label) {
+                              acc += 1;
+                          }
+                          return acc;
+                      }, 0),
+                  );
+              });
+
+              //Timeline data
+              const groupByDate = R.groupBy((response: { response: string; date: string }) => {
+                  const day = dayjs(response.date).format('DD/MM/YY');
+                  return day;
+              });
```

Comment:
No need to have a variable and then return it directly. We can rewrite it to
```suggestion
              const groupByDate = R.groupBy((response: { response: string; date: string }) => dayjs(response.date).format('DD/MM/YY'););
```